### ISSUE TITLE
`@mantine/styles` missing `@emotion/*` dependencies

### ISSUE THREAD

**Issue by MariaSolOs** [mantinedev/mantine#2846]
### What package has an issue

@mantine/core

### Describe the bug

*The actual problem seems to be in `@mantine/styles` but the selection dropdown doesn't have that option*.

In a project using `yarn` 3.2.4, I encounter the following problem when creating a custom styles cache:
<img width="737" alt="image" src="https://user-images.githubusercontent.com/62502207/198890786-cfd43ba0-6a40-4f5b-af2b-0e9178899e45.png">

This will happen even without getting any `yarn` installation errors. After some investigation, I believe this is because `createEmotionCache` is re-exported in `@mantine/core` from `@emotion/cache`, although the former isn't re-exporting those type definitions nor listing the `emotion` library as a dependency.

I'm not sure of the intention behind [these changes](https://github.com/mantinedev/mantine/commit/a93abef62e536233ea87d0fdf320a81a936490ac), but including `@emotion/cache` as a dependency of `@mantine/styles` fixes the issue.

### What version of @mantine/hooks page do you have in package.json?

^5.6.3

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Add `@emotion/cache` to the dependencies of `@mantine/styles`

**Comment by MariaSolOs** [comment]
As a follow-up, it seems like the dependencies on `@emotion/serialize` and `@emotion/utils` should also be added, else `yarn` displays the following "module not found errors":
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/62502207/198898094-eca1ccab-acda-4abe-b376-f1ae1fbcb2d1.png">

Above I had already fixed the issue with `@emotion/serialize` by using the following `packageExtension`:
```
"@mantine/styles@5.6.3":
    dependencies:
      "@emotion/cache": "^11.10.5"
      "@emotion/serialize": "^1.1.1"
```

**Comment by MariaSolOs** [comment]
Is there any extra information that I could provide?

**Comment by rtivital** [comment]
No, these changes can be breaking for some package managers, for now, we just do not support yarn v3

**Comment by MariaSolOs** [comment]
@rtivital I see. Well once you do I would be more than happy to contribute with that, since I already have the workarounds to make it work locally.
I'll close this as won't fix. Thank you!

**Comment by MariaSolOs** [comment]
After some thought I've decided to reopen this so we make sure it gets fixed when merging #2842 (or performing the corresponding Mantine update).

**Comment by MariaSolOs** [comment]
For future reference:
- Why this is currently working with older versions of `yarn`: https://yarnpkg.com/features/pnp#fallback-mode 
- Relevant StackOverflow threads: https://stackoverflow.com/questions/54392809/how-do-i-handle-optional-peer-dependencies-when-publishing-a-typescript-package/70096366#70096366,

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake and it is still an issue in version 7.0, please reopen it.
